src/controllers/timeEntryController.js

```diff
@@ -1392,22 +1389,39 @@ const timeEntrycontroller = function (TimeEntry) {
       await Promise.all(recalculationPromises);
 
       await session.commitTransaction();
-      clearInterval(keepAliveInterval);
-      res.write('finished the recalculation for hoursByCategory for all users\n');
-      return res.end();
+      // recalculateTangibleStatus = 'completed';
+      recalculationTasks[taskId] = 'Completed';
```

Comment:
Could maybe add a completion time, but that's not needed

Reply:
Thank you for the code review! In the latest commit, I have renamed the variable, and used a queue for storing task status, recording both start and completion times.